packages/contracts/contracts/L1/L2OutputOracle.sol

```diff
@@ -117,6 +120,20 @@ contract L2OutputOracle is Ownable {
         emit l2OutputAppended(_l2Output, _l2timestamp);
     }
 
+    /**
+     * @notice Deletes the most recent output.
+     * @param _l2Output The value of the most recent output. Used to prevent erroneously deleting
+     *  the wrong root
+     */
+    function deleteL2Output(bytes32 _l2Output) external onlyOwner {
+        bytes32 outputToDelete = l2Outputs[latestBlockTimestamp];
+        require(_l2Output == outputToDelete, "Can only delete the most recent  output.");
+        emit l2OutputDeleted(outputToDelete, latestBlockTimestamp);
+
+        delete l2Outputs[latestBlockTimestamp];
+        latestBlockTimestamp = latestBlockTimestamp - submissionInterval;
+    }
```

Comment:
Compared to the [original approach](https://gist.github.com/karlfloersch/02ce7938000680ce7820f39ba06bd9b1#file-staterootoracle-sol-L53-L60), this takes an output instead of a timestamp. I found the original a bit more error prone as there were more edge cases (ahead/behind the tip, not a multiple of the `submissionInterval`). I could also provided no arguments and just deleted the tip, but that seemed like it could lead to race conditions.

Reply:
Generally looks good to me, this isn't a hot code path so I don't see a need for using `checked`. This function does not provide the ability to roll back an arbitrary number of outputs less than the fault proof window old, so we will either need to implement that functionality as well or put this contract behind a proxy so that functionality can be added in the future.

I think we should put this behind a delegatecall proxy for now

Reply:
Since we'll only be checkpointing every 30 minutes or so, do you think we're likely to need that ability? 

Also, for this contract I'm open to a delegate proxy, but I could also imagine we just upgrade via social consensus (ie. to a new address).